src/richie/apps/courses/models/course.py

```diff
@@ -473,27 +530,111 @@ class CourseRun(BasePageExtension):
+                    master=self, language_code=translation_object.language_code
+                ).values_list("pk", flat=True)[0]
+            except IndexError:
+                translation_object.pk = None
+            else:
+                translation_object.pk = target_pk
+            translation_object.master = self
+            translation_object.save()
 
     # pylint: disable=arguments-differ
     def save(self, *args, **kwargs):
         """
         Enforce validation each time an instance is saved.
+        Mark the related course page as dirty if the course run has changed so that the
+        modifications can be published.
         """
+        if self.pk:
+            try:
+                public_instance = self.__class__.objects.get(
+                    draft_course_run__pk=self.pk
+                )
+            except self.__class__.DoesNotExist:
+                pass
+            else:
+                is_visible = (
+                    self.state["priority"] < 6 or public_instance.state["priority"] < 6
+                )
+                # Mark the related course page dirty if the course run content has changed
+                # Break out of the for loop as soon as we found a difference
+                for field in self._meta.fields:
+                    if field.name == "direct_course":
+                        if (
+                            public_instance.direct_course.draft_extension
+                            != self.direct_course
+                            and is_visible
+                        ):
+                            self.direct_course.extended_object.title_set.update(
+                                publisher_state=PUBLISHER_STATE_DIRTY
+                            )  # mark target page dirty in all languages
+                            page = (
+                                public_instance.direct_course.draft_extension.extended_object
+                            )
+                            page.title_set.update(
+                                publisher_state=PUBLISHER_STATE_DIRTY
+                            )  # mark source page dirty in all languages
+                            break
+                    elif (
+                        field.editable
+                        and not field.auto_created
+                        and getattr(public_instance, field.name)
+                        != getattr(self, field.name)
+                        and is_visible
+                    ):
+                        self.direct_course.extended_object.title_set.update(
+                            publisher_state=PUBLISHER_STATE_DIRTY
+                        )  # mark page dirty in all languages
+                        break
+        else:
+            # This is a new instance, mark page dirty in all languages unless
+            # the course run is yet to be scheduled (hidden from public page in this case)
+            if self.state["priority"] < 6:
```

Comment:
This priority of `6` seems important (seen in may lines of code), should it be hard-coded or should we better define (or import) a setting or a more explicit variable for that?

Reply:
Yes! You're right it sucks. I should use constants for that.